main/task.py

```diff
@@ -27,7 +30,35 @@ def send_mail_notification(subject, body, to=None, **kwargs):
       '#####################################################################',
       sender, to or sender, subject, body
     )
-  deferred.defer(mail.send_mail, sender, to or sender, subject, body, **kwargs)
+  if config.CONFIG_DB.mailgun_api_key and config.CONFIG_DB.mailgun_api_base_url:
+    deferred.defer(send_mailgun_message, sender, to or sender, subject, body)
+  else:
+    deferred.defer(mail.send_mail, sender, to or sender, subject, body, **kwargs)
+
+
+def send_mailgun_message(sender, to, subject, body):
+  mailgun_api_key = config.CONFIG_DB.mailgun_api_key
+  mailgun_api_base_url = config.CONFIG_DB.mailgun_api_base_url
+  url = '%s/messages' % mailgun_api_base_url
+  data = {'from': sender,
+    'to': to,
+    'subject': subject,
+    'text': body}
```

Comment:
```python
  data = {
    'from': sender,
    'to': to,
    'subject': subject,
    'text': body,
  }
```
or that since it can fit in one line :)

```python
  data = {'from': sender, 'to': to, 'subject': subject, 'text': body}
```